src/test/java/org/testng/xml/XmlSuiteTest.java

```diff
@@ -71,4 +71,17 @@ public void ensureNoExceptionsAreRaisedWhenMethodSelectorsDefinedAtSuiteLevel()
         assertThat(xmlsuite.toXml()).isNotEmpty();
     }
 
+    @Test(description = "GITHUB-1674")
+    public void ensureSuiteLevelBeanshellIsAppliedToAllTests() throws IOException {
+        Parser parser = new Parser("src/test/resources/xml/issue1674.xml");
+        List<XmlSuite> suites = parser.parseToList();
+        XmlSuite xmlsuite = suites.get(0);
+        assertThat(xmlsuite.getTests().get(0).getMethodSelectors().size()).isEqualTo(0);
+        TestNG testNG = create();
+        testNG.setXmlSuites(suites);
+        testNG.setUseDefaultListeners(false);
+        testNG.run();
+        assertThat(xmlsuite.getTests().get(0).getMethodSelectors().size()).isEqualTo(1);
```

Comment:
Is there any way to assert that the BeanShell code actually ran instead of just testing this?

Reply:
@cbeust - I added an assertion that will check to see if the beanshell got executed. Can you please check if thats fine ?

Reply:
Looks good to me, feel free to merge if you're happy with it.

Thanks!